src/mio_wrapper.rs

```diff
@@ -0,0 +1,111 @@
+use std::io;
+use std::os::unix::io::{RawFd, FromRawFd};
+
+use libc::{self, c_void};
+use mio::{Poll, Token, Ready, PollOpt, Evented};
+use mio::unix::EventedFd;
+
+pub struct Tun(RawFd);
+pub struct Tap(RawFd);
```

Comment:
Also, since most of the code is the same, you could do two wrappers around a common implementation.